src/apps/shared/test/client.py

```diff
@@ -9,31 +11,85 @@ class TestClient:
     def __init__(self):
         app = create_app()
         self.client = AsyncClient(app=app, base_url="http://test.com")
+        self.headers = dict()
 
     @staticmethod
     def _prepare_url(url, query):
         return f"{url}{urllib.parse.urlencode(query)}"
 
-    async def post(self, url, query=None, data=None, headers=None) -> Response:
+    def _get_headers(self, headers: Optional[dict] = None) -> dict:
+        headers_ = dict(self.headers)
+        if headers:
+            headers_.update(headers)
+        return headers_
```

Comment:
I think this method is doing not the declared functionality.

It updates headers not only _get_headers()

Reply:
Done